### DISCUSSION TITLE
How to handle wildcard domain request

### DISCUSSION THREAD

**Discussion by Jeyanthinath** [pocketbase/pocketbase#3164] [Q&A]
I want to create a subdomain for each user logged into my application and they will create a public page 
            (eg :    User can login to my app and create a subdomain `test.mywebsite.com`)

-  If he is logged in I can use the session and figure out which site he belongs to and serve the content
- If the user publishes his site online as public, it will be available at `test.mywebsite.com` for any non logged-in users to use, using a normal python or Golang code I can see if the request is coming from some subdomain (in this case `test`), but with Pocketbase I don't see any option here.

Can someone help me on this ?

(I don't want to manage or spin up a new instance for each new user)

**Comment by ganigeorgiev** [comment]
In general you can access the host address in a middleware or route with `c.Request().Host` and extract the subdomain from it, but I'm not sure that I understand the requirements to provide a more complete example.

Could you share how you would do it if it was regular Python or Go code?

**Comment by Jeyanthinath** [comment]
My use case:

- user can login to my website and create a new subdomain for their blog
- the user can publish the blog to public

eg: (user can login to my site `test.com` ,and they can create a subdomain called `myblog.test.com` and expose it to public internet)

Likewise multiple users can create their own site (eg: `anotherblog.test.com` , `example.test.com`) . Now I want to show the website based on their sub-domain.

Eg: If the user hits  the public site
(myblog.test.com -> load home page content created by the user 1)
(example.test.com -> load the homepage content created by the user 3, as he is the owner of site example)

How to handle this scenario in the pocket base, as the user vising `myblog.test.com` without logging in , so no user session details coming . 

eg source code in python

```from flask import Flask

app = Flask(__name__)

@app.route('/')
def base_route():
    return "Non sub-domain conent"

@app.route('/', subdomain='<subdomain>')
def subdomain_route(subdomain):
    if subdomain == 'myblog':
        return "Hey, this content is posted by user 1"
    elif subdomain == 'example':
        return "This is content for the user 2"
```

**Reply by ganigeorgiev** [reply]
Here is a very rough example:

```go
package main

import (
	"log"
	"net/http"
	"strings"

	"github.com/labstack/echo/v5"
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/apis"
	"github.com/pocketbase/pocketbase/core"
)

func main() {
	app := pocketbase.New()

	app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		e.Router.GET("/*", func(c echo.Context) error {
			host := c.Request().Host

			// just get the first part of the domain
			// (note that this could return also the main domain)
			prefix := strings.Split(host, ".")[0]

			switch prefix {
			case "myblog":
				return c.String(http.StatusOK, "myblog...")
			case "example":
				return c.String(http.StatusOK, "example...")
			}

			return apis.NewNotFoundError("", nil)
		})

		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```

_But keep in mind that the subdomain extraction, in the above `prefix`, may not handle all cases (eg. should `www.sub.example.com` be treated the same as `sub.example.com`?) and instead you could use a dedicated external Go package that will properly parse the domain or implement your own more advanced handling._

**Reply by ganigeorgiev** [reply]
Also note that you can register a plain Go net/http handler using `echo.WrapHandler(h http.Handler)`.

**Reply by Jeyanthinath** [reply]
Thanks for the details answer

Adding to the things on top,  I need to form a query to the pocketbase based on which subdomain someone is accessing the site.

eg: 

if someone access from `sub-domain.test.com`, the query should be `select * from the blogs where the domain="sub-domain"` , currently I plan on handling this directly on svelte server side, rather than handling on pocketbase.
